clippy_lints/src/tabs_in_doc_comments.rs

```diff
@@ -104,30 +104,29 @@ fn get_chunks_of_tabs(the_str: &str) -> Vec<(u32, u32)> {
     // tracker to decide if the last group of tabs is not closed by a non-tab character
     let mut is_active = false;
 
-    let chars_array: Vec<_> = the_str.chars().collect();
+    let char_indices: Vec<_> = the_str.char_indices().collect();
 
-    if chars_array == vec!['\t'] {
+    if char_indices.len() == 1 && char_indices.first().unwrap().1 == '\t' {
```

Comment:
Can you use pattern matching here?

```suggestion
    if let &[(_, '\t')] = &char_indices {
```

+/- compiler/type errors.

Reply:
Yes, that works! :exploding_head: